backend/drm/atomic.c

```diff
@@ -195,9 +196,55 @@ static bool atomic_crtc_move_cursor(struct wlr_drm_backend *drm,
 	return atomic_end(drm->fd, &atom);
 }
 
+static bool atomic_crtc_set_gamma(struct wlr_drm_backend *drm,
+		struct wlr_drm_crtc *crtc, uint16_t *r, uint16_t *g, uint16_t *b,
+		uint32_t size) {
+	struct drm_color_lut *gamma = calloc(sizeof(struct drm_color_lut), size);
```

Comment:
Allocate this on the stack

Reply:
Do you request the use of VLAs? (the size of the allocated object is `sizeof(struct drm_color_lut) * size`)

Reply:
Yes

Reply:
Also make all these pointers const please

Reply:
Making these pointers const would require the interface to have const r,g and b, which is impossible since `drmModeCrtcSetGamma` uses non-const r, g and b. It requires a cast when filling out the interface if we still want to do this.

Reply:
We can change the interface, too.

Reply:
Well no, since then we can't call `drmModeCrtcSetGamma` because it has non constant r, g and b parameters.

Reply:
:(